ironfish/src/wallet/wallet.ts

```diff
@@ -957,7 +903,7 @@ export class Wallet {
 
     if (noteMapValue) {
       this.logger.debug(`Unspent note has index ${String(noteMapValue.index)}`)
-      await sender.updateDecryptedNote(unspentNote.hash, {
+      await this.walletDb.saveDecryptedNote(sender, unspentNote.hash, {
```

Comment:
Same for this change.. We already do an early return if there are no nullifiers so no need to check deletion of nullifier here either?

Reply:
Yeah we know that the note has a nullifier already, so we wouldn't delete the index entry in this case